### DISCUSSION TITLE
82. DROP TABLE과 TRUNCATE TABLE의 차이를 설명해 주세요.

### DISCUSSION THREAD

**Discussion by cutehumanS2** [jewelry-box/cs-interview#84] [General]
- 테이블에 AUTO_INCREMENT를 사용하는 경우, DROP TABLE과 TRUNCATE TABLE을 실행했을 때 ID 값은 어떻게 동작하나요?

**Comment by linirini** [comment]
- DROP TABLE
  - 테이블의 구조(메타데이터)와 데이터 모두 삭제
  - 테이블이 삭제되므로, 이후에는 테이블 자체를 사용할 수 없다.
  - ID는 테이블을 재생성해야 초기화된다.
- TRUNCATE TABLE
  - 테이블의 데이터만 삭제하고 테이블 구조와 테이블의 열, 제약 조건, 인덱스 등은 그대로 남는다.
  - 테이블에 ID 열이 포함되어 있으면 해당 열의 카운터는 열에 대한 초기값으로 다시 설정

**Reply by cutehumanS2** [reply]
리니 오타 수정해 주세요~

**Reply by cutehumanS2** [reply]
DROP TABLE의 3번 항목 보충입니다
> AUTO_INCREMENT 열의 값도 삭제되며, 테이블을 재생성해야 초기값으로 돌아갑니다.

TRUNCATE TABLE의 2번 항목 보충할게요.

>  테이블에 AUTO_INCREMENT 열이 포함되어 있다면, 해당 열의 카운터는 초기값으로 재설정됩니다.
